repair/repair.cc

```diff
@@ -964,6 +964,16 @@ static future<> do_repair_ranges(lw_shared_ptr<repair_info> ri) {
             });
         });
 
+        if (ri->reason != streaming::stream_reason::repair) {
+            try {
+                auto& table = ri->db.local().find_column_family(table_id);
+                rlogger.debug("repair[{}]: Trigger off-strategy compaction for keyspace={}, table={}",
+                    ri->id.uuid, table.schema()->ks_name(), table.schema()->cf_name());
+                table.trigger_offstrategy_compaction();
+            } catch (replica::no_such_column_family&) {
+                // Ignore dropped table
```

Comment:
will it work for decommission too?

Reply:
The decommission is more complicated because it needs the node running decommission to tell peer nodes that the ranges are repaired. I am not solving it in this series yet.

Reply:
I am asking this because we had to piggyback on decommission_done for triggering off-strategy on nodes receiving data. Then perhaps we'll need a new node_ops_cmd.

Reply:
> The decommission is more complicated because it needs the node running decommission to tell peer nodes that the ranges are repaired. I am not solving it in this series yet.

Makes sense to postpone it. Definitely shouldn't block this series.

Reply:
We can reuse decommission_done, by specifying the table uuid that is finished. But this will ask repair to send node ops cmd. I am still thinking about if we could have a better solution.